### PR TITLE
404s with formats other than html

### PR DESCRIPTION
**Pull request by banukutlu**
#294

### CODE REVIEW

**Comment by cdmo** [comment]
I'm still getting a `500` when I try to go to http://localhost:3001/wcd/system.xml.

**Review by cdmo — CHANGES_REQUESTED** [review]
Needs to be that unauthenticated traffic going to http://localhost:3001/wcd/system.xml should experience a 404 or perhaps better would be a 401 Unauthenticated

**Comment by banukutlu** [comment]
> Needs to be that unauthenticated traffic going to http://localhost:3001/wcd/system.xml should experience a 404 or perhaps better would be a 401 Unauthenticated

Could you share a screenshot or logs? When going there I am seeing the 404 page 🤔

**Comment by banukutlu** [comment]
> Needs to be that unauthenticated traffic going to http://localhost:3001/wcd/system.xml should experience a 404 or perhaps better would be a 401 Unauthenticated

looking into it 👍  I missed to check unauthenticated requests

**Review by cdmo — CHANGES_REQUESTED** [review]
When unauthenticated, I go to http://localhost:3001/wcd/system.xml and I am getting an error `undefined method `[]' for nil:NilClass` see [Screen Shot 2020-11-12 at 9 29 08 AM](https://user-images.githubusercontent.com/523381/98952843-edb4af00-24c9-11eb-898f-50ff199d231d.png)

Also, I am unable to login into the application. 

🤔

**Comment by banukutlu** [comment]
> When unauthenticated, I go to http://localhost:3001/wcd/system.xml and I am getting an error `undefined method `[]' for nil:NilClass` see [Screen Shot 2020-11-12 at 9 29 08 AM](https://user-images.githubusercontent.com/523381/98952843-edb4af00-24c9-11eb-898f-50ff199d231d.png)
> 
> Also, I am unable to login into the application.
> 
> 🤔

made some changes, rebased from master to get that patron validation to use that to fix this issue, it seems to be working on my local, could you give it a try again?

File: app/controllers/errors_controller.rb [hunk 1]
```diff
@@ -2,7 +2,10 @@
 
 class ErrorsController < ApplicationController
   def not_found
-    render status: 404
+    respond_to do |format|
+      format.html { render status: 404 }
+      format.all { render status: 404, formats: :html, content_type: 'text/html' }
```
Comment (cdmo) [inline app/controllers/errors_controller.rb:7]:
```suggestion
      format.any { render status: 404, formats: :html, content_type: 'text/html' }
```

"Respond to also allows you to specify a common block for different formats by using any:" [respond_to](https://apidock.com/rails/ActionController/MimeResponds/respond_to)

this seems to work, did you try it before and it didn't?

**Review by cdmo — CHANGES_REQUESTED** [review]
We are considering forcing authentication on the `ErrorsController` which essentially completely seals off the application to public traffic.

**Review by cdmo — APPROVED** [review]
LGTM! 🎸

**Pull request merged**